TicketingService.Monitoring/Program.cs

```diff
@@ -0,0 +1,156 @@
+        });
+    }
+
+    public static async Task<string> GroupByRegistry(IDocumentStore store, CancellationToken cancellationToken)
+    {
+        return JsonConvert.SerializeObject("Not implemented");
+    }
+
+    private static async Task<IReadOnlyList<Ticket>> GetCompletedTickets(this IDocumentStore store, CancellationToken cancellationToken)
+    {
+        await using var session = store.QuerySession();
+        return await session.Query<Ticket>()
+            .Where(t => t.Status == TicketStatus.Complete && t.Created > _fromDate)
+            .ToListAsync(cancellationToken);
+    }
+}
+
+public static class Extensions
+{
+    public static WebApplicationBuilder AddAppSettings(this WebApplicationBuilder builder, string[] args)
+    {
+        builder.Configuration
+        .AddJsonFile("appsettings.json")
+        .AddJsonFile($"appsettings.{builder.Environment.EnvironmentName.ToLowerInvariant()}.json", optional: true, reloadOnChange: false)
+        .AddJsonFile($"appsettings.{Environment.MachineName.ToLowerInvariant()}.json", optional: true, reloadOnChange: false)
+        .AddEnvironmentVariables()
+        .AddCommandLine(args);
+
+        return builder;
+    }
+
+    public static WebApplicationBuilder AddServices(this WebApplicationBuilder builder)
+    {
+        var options = builder.GetAppOptions<AppOptions>();
+
+        builder.Services.AddEndpointsApiExplorer();
+        builder.Services.AddSwaggerGen();
+        builder.Services
+            .AddMartenTicketing(options.ConnectionString)
+            .AddHealthChecks().AddNpgSql(_ => options.ConnectionString);
+
+        return builder;
+    }
+
+    public static WebApplicationBuilder AddOptions<TOptions>(this WebApplicationBuilder builder)
+         where TOptions : class
+    {
+        builder.Services.Configure<TOptions>(builder.Configuration.GetSection(typeof(TOptions).Name));
+        return builder;
+    }
+
+    public static TOptions GetAppOptions<TOptions>(this WebApplicationBuilder builder)
+        where TOptions: class, new()
+    {
+        var options = new TOptions();
+        builder.Configuration.Bind(typeof(TOptions).Name, options);
+
+        var requiredProperties = options
+            .GetType()
+            .GetProperties()
+            .Where(prop => Attribute.IsDefined(prop, typeof(RequiredAttribute)));
+
+        void ThrowArgumentNullException(PropertyInfo p) => throw new ArgumentNullException($"{typeof(TOptions).Name}.{p.Name}");
+
+        foreach (var prop in requiredProperties)
+        {
+            var obj = prop.GetValue(options, null);
+
+            if (prop.GetValue(options, null) is null)
+            {
+                ThrowArgumentNullException(prop);
+            }
+
+            if (obj is string valstr && string.IsNullOrEmpty(valstr))
+            {
+                ThrowArgumentNullException(prop);
+            }
+        }
+
+        return options;
+    }
+}
+
+public class AppOptions
+{
+    [Required]
```

Comment:
there is now a required keyword that you can set in front of a property type: 
`public required string ConnectionString { get; set; }`